tests/plugins/imagebase/features/caption.js

```diff
@@ -0,0 +1,334 @@
+/* bender-tags: editor,widget */
+/* bender-ckeditor-plugins: imagebase,toolbar */
+/* bender-include: ../../widget/_helpers/tools.js */
+/* global widgetTestsTools */
+
+( function() {
+	'use strict';
+
+	bender.editors = {
+		classic: {},
+
+		divarea: {
+			config: {
+				extraPlugins: 'divarea'
+			}
+		},
+
+		inline: {
+			creator: 'inline'
+		}
+	};
+
+	function addTestWidget( editor ) {
+		if ( editor.widgets.registered.testWidget ) {
+			return;
+		}
+
+		var plugin = CKEDITOR.plugins.imagebase;
+
+		plugin.addImageWidget( editor, 'testWidget', plugin.addFeature( editor, 'caption', {} ) );
+	}
+
+	function getFixture( name ) {
+		return CKEDITOR.document.getById( name ).getHtml();
+	}
+
+	function assertVisibility( caption, isVisible, msg ) {
+		assert[ 'is' + ( isVisible ? 'False' : 'True' ) ]( !!caption.data( 'cke-hidden' ), msg );
+	}
+
+	function createToggleTest( options ) {
+		return function( editor, bot ) {
+			addTestWidget( editor );
+
+			bot.setData( getFixture( options.fixture ), function() {
+				var widget = widgetTestsTools.getWidgetByDOMOffset( editor, 0 ),
+					caption = widget.parts.caption,
+					focusTrap = editor.editable().findOne( 'p' ).getChild( 0 ),
+					range = editor.createRange();
+
+				assertVisibility( caption, options.initial, 'caption visibility (initial)' );
+
+				if ( options.onInit ) {
+					options.onInit( widget );
+				}
+
+				if ( options.customFocus ) {
+					options.customFocus( widget );
+				} else {
+					widget.focus();
+				}
+
+				assertVisibility( caption, options.focus, 'caption visibility (focus)' );
+
+				if ( options.onFocus ) {
+					options.onFocus( widget );
+				}
+
+				range.setStart( focusTrap, 1 );
+				range.collapse();
+				range.select();
+
+				assertVisibility( caption, options.blur, 'caption visibility (blur)' );
+
+				if ( options.onBlur ) {
+					options.onBlur( widget );
+				}
+			} );
+		};
+	}
+
+	function createMultipleToggleTest( options ) {
```

Comment:
Please, add at least minimal comment to what the function does/what is its purpose. Currently one has to read its source code to understand this.

Reply:
Please, review this assertion. It's a cause for `'test toggling captions (two widgets with empty captions)'` to fail on IE11.

Reply:
@mlewand which assertion exactly are you thinking of?

Reply:
Actually I was referring to `createMultipleToggleTest` helper.